agents/scripts/core/lock-claim.sh

```
@@ -65,16 +65,18 @@ remote="${LOCK_REMOTE:-origin}"
-    # path-or-host segments. Still permissive enough for legitimate forks
-    # (`Smatchet-fork`, `smatchet-experiment`).
-    case "$remote_url" in
-        *[/:][Ss]matchet*) : ;;
-        *) echo "lock-claim: remote URL '$remote_url' does not look like a Smatchet repo (set SMATCHET_LOCK_BYPASS_REPO_CHECK=1 to override)" >&2; exit 2 ;;
+    # Portability: derive the expected repo identifier from project.config.json
+    # (project.name) instead of hardcoding "Smatchet", so the portable agents/
+    # tree works in another project. Falls back to "Smatchet" when the config is
+    # unreadable, so the guard is never weaker than before. Anchored at a path
+    # boundary (`/` https, `:` scp-style) per H15; matched case-insensitively.
+    _proj_cfg="$(git rev-parse --show-toplevel 2>/dev/null)/project.config.json"
+    _proj_name="$( { python3 -c 'import json,sys;print(json.load(open(sys.argv[1]))["project"]["name"])' "$_proj_cfg" 2>/dev/null \n                    || python -c 'import json,sys;print(json.load(open(sys.argv[1]))["project"]["name"])' "$_proj_cfg" 2>/dev/null; } || printf 'Smatchet' )"
```

Shellcheck (0.11.0)

[warning] \n is just literal 'n' here. For line feed, use a quoted, literal line feed instead.

(SC1012)